OutOfSchool/OutOfSchool.BusinessLogic/Services/Database/WorkshopService.cs

```diff
@@ -899,4 +893,29 @@ private async Task UpdateWorkshop()
     }
 
     private void ValidateOffsetFilter(OffsetFilter offsetFilter) => ModelValidationHelper.ValidateOffsetFilter(offsetFilter);
+
+    private async Task UpdateWorkshopStatusBySeatsLimitAndAvailability(uint newAvailableSeats, Workshop currentWorkshop)
+    {
+        var currentWorkshopTakenSeats = currentWorkshop.Applications.TakenSeats();
```

Comment:
(Logic) Related to the whole method. Here you change workshop status when some seats-related conditions are met. But you don't check the old workshop status. Thus you could set the same value for it again and again. Isn't it better to do changes only if they are really necessary?

Reply:
Thanks for the change! 
One minor code style question - could you merge nested `if`s into one `if`?

Reply:
Fixed. Thank you!